src/api/middleware/validators/company.js

```diff
@@ -75,6 +75,10 @@ export const deleteCompany = useExpressValidators([
     existingCompanyParamValidator,
 ]);
 
+export const profile = useExpressValidators([
+    existingCompanyParamValidator,
+]);
```

Comment:
This means that when a company does not exists we will return a HTTP 422 error, instead of HTTP 404 (NOT FOUND) https://github.com/NIAEFEUP/nijobs-be/blob/d4dfefa378063d3172c05810977876b0216a3420/src/api/middleware/errorHandler.js#L57

We should not have this check on the validators, but on the route code itself, and throw the error with the correct status there (similar to what is done [here](https://github.com/NIAEFEUP/nijobs-be/blob/ca3c0e98c65297353e2bf74e5497f8cb82b22238/src/api/middleware/files.js#L30-L39))

Reply:
I understand returning 404 instead of 422, but by this line of thought shouldn't every other route that has a `companyId` request parameter have the check on the route code and not on the validators ?